### DISCUSSION TITLE
simple_cycles speed up

### DISCUSSION THREAD

**Discussion by giovannamagri** [networkx/networkx#7577] [Q&A] [answered]
Dear all,
For my thesis I'm currently in need to find all the cycles of different length in my network for different aggregation periods (monthly, quarterly, yearly). I managed to find everything at the daily level, but by aggregating the data, the number of interactions across nodes exponentially increases as well as the number of cycles. Hence, my code repeatedly crashes when asking to find them and store them according to their length. Do you have any suggestion on the matter? I would be infinitely grateful.

**Comment by dschult** [comment, marked as answer]
What do you mean by "crashes"?
If you are running out of memory, you might try splitting up the network so you aren't trying to hold everything in memory at once. This is a hard problem. The amount of memory needed increases very fast as the number of edges increases.
